frontend/src/utils/index.js

```diff
@@ -571,22 +571,33 @@ export function generateWorker (availableZones, cloudProfileName, region) {
     maximum: 2,
     maxSurge: 1,
     machine: {
-      type: machineType,
+      type: machineType.name,
       image: machineImage
     },
     zones,
     isNew: true
   }
-  if (volumeType) {
+  if (volumeType.name) {
+    worker.volume = {
+      type: volumeType.name,
+      size: defaultVolumeSize
+    }
+  } else if (machineType.storage && !machineTypeHasStorageWithTypeFixed(machineType)) {
+    worker.volume = {
+      size: machineType.storage.size
+    }
+  } else if (!machineType.storage) {
```

Comment:
move this else if before `else if (machineType.storage && !machineTypeHasStorageWithTypeFixed(machineType)) `

Reply:
done